types/@ember-decorators/utils/decorator.d.ts

```diff
@@ -0,0 +1,14 @@
+type MemberDescriptor = {
+  key: string;
+  kind: 'class' | 'method' | 'field' | 'initializer';
+  elements?: MemberDescriptor[];
+  key: string;
+  placement?: 'prototype' | 'static' | 'own';
+  extras?: MemberDescriptor[];
+  initializer?: () => any;
+  descriptor?: PropertyDescriptor;
+};
+
+export function decoratorWithParams(
+  fn: (desc: MemberDescriptor, params?: any[]) => MemberDescriptor
+): any;
```

Comment:
This is a bit strange from a TS decorator standpoint. Is `any` really the return type we want here?

Also, are we sure that this

```ts
  initializer?: () => any;
```
shouldn't be 
```ts
  initializer?: () => void;
```

Reply:
> Is `any` really the return type we want here?

It is the return type that was used before:

https://github.com/rwjblue/sparkles-component/blob/e320c6ffba4681fe3cbcb74abcbe24aad7bffef7/addon/tracked.ts#L121-L126

And it does not throw any type errors. Actually `PropertyDescriptor` also does not throw any type errors. I don't know where I got that from, because all the official resources use `void`.

So I guess, I'm gonna switch. Thanks. 👍 	


> Also, are we sure that this
> 
> ```ts
>   initializer?: () => any;
> ```
> 
> shouldn't be
> 
> ```ts
>   initializer?: () => void;
> ```

Yes, because for field initializers the return value is used as the initial value of that field:

https://github.com/rwjblue/sparkles-component/blob/e320c6ffba4681fe3cbcb74abcbe24aad7bffef7/addon/tracked.ts#L76-L86

Quoting from [`TABLE.md`](https://github.com/tc39/proposal-decorators/blob/master/TABLE.md):

> <sup>3</sup> when `kind` is `field` don't include the value into the `descritor.value` and the value, if exist, is returned by `initializer` function.